### ISSUE TITLE
Pull request #24 (Introduction of AttemptClass and PaymentClass) broke the payment algorithm or simulation

### ISSUE THREAD

**Issue by adamritter** [renepickhardt/pickhardtpayments#33]
It seems like this merged pull request broke the payment algorithm or simulation:

https://github.com/renepickhardt/pickhardtpayments/pull/24

PickhardtPaymentsExample.ipynb examples don't work anymore:

Round number: 10
Try to deliver 18879883 satoshi:

Statistics about 46 candidate onions:

successful attempts:

p = 100.00% amt: 216543 sats hops: 4 ppm: 1413
p = 94.37% amt: 84263 sats hops: 5 ppm: 1349
p = 90.69% amt: 59164 sats hops: 5 ppm: 1708
p = 75.00% amt: 337453 sats hops: 5 ppm: 1417
p = 73.91% amt: 112351 sats hops: 5 ppm: 1374
p = 99.52% amt: 42718 sats hops: 6 ppm: 1348
p = 100.00% amt: 35946 sats hops: 6 ppm: 1367
p = 94.97% amt: 242846 sats hops: 7 ppm: 1696
p = 77.42% amt: 2351 sats hops: 9 ppm: 1427

failed attempts:

p = 100.00% amt: 88000 sats hops: 2 ppm: 1339
p = 74.25% amt: 288187 sats hops: 2 ppm: 1341
p = 14.57% amt: 1574652 sats hops: 2 ppm: 1359
p = 100.00% amt: 102400 sats hops: 2 ppm: 1381
p = 100.00% amt: 102400 sats hops: 2 ppm: 1356
....

SUMMARY:

Rounds of mcf-computations: 10
Number of attempts made: 448
Number of failed attempts: 356
Failure rate: 79.46%
total Payment lifetime (including inefficient memory management): 8.661 sec
Learnt entropy: 229.60 bits
fee for settlement of delivery: 0.000 sat --> 0 ppm
used mu: 100

**Comment by renepickhardt** [comment]
yes we introduced a bug but that was supposed to be fixed with the merge of #18 which is currently on my desk for review. Sorry for the slight delay that has been introduced to this repository over the summer months.

**Comment by adamritter** [comment]
No worries :)
